File: languages/django_models/institution.py
from .abstract_element_model import AbstractElementModel
from django.db import models
from ow.elements.models.object import Object as ObjectModel

class Institution(AbstractElementModel):

    # Foundation
    doctrine = models.TextField(blank=True, null=True)
    founding_date = models.PositiveIntegerField(blank=True, null=True)
    parent_institution = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="institution_parent_institution")

    # Claims
    zones = models.ManyToManyField("Zone", blank=True, related_name="institution_zones")
    objects = models.ManyToManyField(ObjectModel, blank=True, related_name="institution_objects")  # type: ignore
    creatures = models.ManyToManyField("Creature", blank=True, related_name="institution_creatures")

    # World
    status = models.TextField(blank=True, null=True)
    allies = models.ManyToManyField("Institution", blank=True, related_name="institution_allies")
    adversaries = models.ManyToManyField("Institution", blank=True, related_name="institution_adversaries")
    constructs = models.ManyToManyField("Construct", blank=True, related_name="institution_constructs")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Institution.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Institution : Element
{
    [JsonProperty("doctrine"), TextAttribute("")]
    public string doctrine;
    [JsonProperty("founding_date"), Integer(0)]
    public int foundingDate;
    [JsonProperty("parent_institution"), ReferenceAttribute(typeof(Institution))]
    public string parentInstitution;
    [JsonProperty("legislation"), ReferenceAttribute(typeof(Law), true)]
    public string legislation;
    [JsonProperty("zones"), ReferenceAttribute(typeof(Zone), true)]
    public string zones;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("status"), TextAttribute("")]
    public string status;
    [JsonProperty("allies"), ReferenceAttribute(typeof(Institution), true)]
    public string allies;
    [JsonProperty("adversaries"), ReferenceAttribute(typeof(Institution), true)]
    public string adversaries;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
}
